File: 1961.py
entrada = input().split()
altura_pulo = int(entrada[0])
canos = int(entrada[1])
gameover = 0
entrada = input().split()

for i in range(len(entrada) - 1):
    cano_atual = int(entrada[i])
    proximo_cano = int(entrada[i + 1])
    
    if (abs(cano_atual - proximo_cano) > altura_pulo):
        gameover = 1
        break
    
if (gameover == 1):
    print("GAME OVER")
else:
    print("YOU WIN")

File: 1961.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

var entrada = lines.shift().trim().split(" ");
var altura_pulo = Number(entrada[0]);
var canos = Number(entrada[1]);
var altura_canos = lines.shift().trim().split(" ");
var gameover = false;

for (var i = 0; i < altura_canos.length - 1; i++) {
    var cano_atual = Number(altura_canos[i]);
    var proximo_cano = Number(altura_canos[i + 1]);
    
    if (Math.abs(proximo_cano - cano_atual) > altura_pulo) {
        gameover = true;
        break;
    }
}

if (!gameover) {
    console.log("YOU WIN");
} else {
    console.log("GAME OVER");
}
